### ISSUE TITLE
`Dense(units=1)` output shape is not `(batch_size, 1)` as expected

### ISSUE THREAD

**Issue by unsignedrant** [keras-team/keras#16650]
Here's a model:

    import random
    import string
    
    import tensorflow as tf
    from keras import Input, Model
    from keras.layers import Dense, Dropout
    from transformers import DistilBertTokenizer, TFDistilBertModel
    
    
    def create_model(input_shape, cache_dir, pretrained_model):
        ids = Input(input_shape, dtype='int32', name='input_ids')
        masks = Input(input_shape, dtype='int32', name='attention_mask')
        x0 = dict(input_ids=ids, attention_mask=masks)
        x = TFDistilBertModel.from_pretrained(pretrained_model, cache_dir=cache_dir)(x0)[0]
        x = Dropout(0.2)(x)
        x = Dense(64)(x)
        x = Dropout(0.2)(x)
        output = Dense(1, 'sigmoid')(x)
        return Model(x0, output)

I need the output to be of shape `(100, 1)` however, it's `(100, 512)` in this example:

    pretrained_model = 'distilbert-base-uncased'
    cache_dir = '.'
    tokenizer = DistilBertTokenizer.from_pretrained(
        pretrained_model, cache_dir=cache_dir
    )
    s = string.ascii_letters
    text = [''.join(random.choice(s) for _ in range(10)) for _ in range(100)]
    tokens = tokenizer.batch_encode_plus(
        text,
        add_special_tokens=True,
        truncation=True,
        return_tensors='np',
        padding='max_length',
    )
    model = create_model((None,), cache_dir, pretrained_model)
    preds = model(dict(tokens))
    print(preds.shape)

shape:

    TensorShape([100, 512])

**Comment by sushreebarsa** [comment]
@unsignedrant 
In order to expedite the trouble-shooting process, please provide a complete code snippet to reproduce the issue reported here. Thank you!

**Comment by unsignedrant** [comment]
What's missing?

**Comment by sushreebarsa** [comment]
@unsignedrant Could you please refer to this [gist](https://colab.research.google.com/gist/sushreebarsa/9dfa12fc6f7e18830add43745329e008/16650.ipynb#scrollTo=zcytXcb2o46x) and confirm the reported issue?
Thank you!

**Comment by unsignedrant** [comment]
confirmed

**Comment by gadagashwini** [comment]
Hi @unsignedrant, `TFDistilBertModel` has **512** default value for `max_position_embeddings `. 
Output of your model prediction is
`(100, 512, 1)`
This is expected output. 
100 is batch size, 512 embedding position.

Thank you!

**Comment by unsignedrant** [comment]
The model is meant to accept text examples and output a rank for each example in the batch. What would be the proper way to do it in this case?

**Comment by gadagashwini** [comment]
Hi @unsignedrant, Output of the (embeddings + one for the output of each layer) of shape `(batch_size, sequence_length, hidden_size).` Thank you!
